src/reduce.test.js

```diff
@@ -11,9 +11,41 @@ describe('reduce', () => {
     delete Array.prototype.reduce2;
   });
 
-  it('should ', () => {
+  it(`should return error if callback doesn't passed`, () => {
+    expect(() => {
+      [1, 2, 3].reduce2();
+    }).toThrow();
+  });
+
+  it('should work with undefined passed as an initial value', () => {
+    const callback = (a, b) => a + b;
+
+    expect([1, 2, 3].reduce2(callback, undefined)).toBe(NaN);
```

Comment:
The expectation `toBe(NaN)` might not work as intended because `NaN` is not equal to itself in JavaScript. Consider using `toBeNaN()` or `Number.isNaN()` for this test.